src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -194,6 +194,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         checkReferenceIsPublic(path, path, holder)
     }
 
+    private fun checkLifetimeParameter(holder: AnnotationHolder, lifetimeParameter: RsLifetimeParameter) {
+        checkReservedLifetimeName(holder, lifetimeParameter)
+        checkDuplicates(holder, lifetimeParameter)
+    }
+
+    private fun checkReservedLifetimeName(holder: AnnotationHolder, lifetimeParameter: RsLifetimeParameter) {
+        val lifetimeName = lifetimeParameter.quoteIdentifier.text
+        when (lifetimeName) {
+            "'_",
+            "'static" -> RsDiagnostic.ReservedLifetimeNameError(lifetimeParameter, lifetimeName).addToHolder(holder)
```

Comment:
Let's extract the list of reserved names to a constant and then just check `if (lifetimeName in RESERVED_LIFETIME_NAMES)`